localstack/services/s3/provider.py

```diff
@@ -1150,6 +1161,109 @@ def get_object_attributes(
 
         return response
 
+    def put_bucket_analytics_configuration(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        id: AnalyticsId,
+        analytics_configuration: AnalyticsConfiguration,
+        expected_bucket_owner: AccountId = None,
+    ) -> None:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        validate_analytics_configuration(analytics_configuration)
+        store = self.get_store()
+        if bucket not in store.bucket_analytics_configuration:
+            store.bucket_analytics_configuration[bucket] = [analytics_configuration]
+            return
+        for analytics_config in store.bucket_analytics_configuration[bucket]:
+            if analytics_config["Id"] == id:
+                analytics_config.update(analytics_configuration)
+                return
+        store.bucket_analytics_configuration[bucket].append(analytics_configuration)
```

Comment:
nit: related to the model change, this could be a bit easier by using a dict to store the bucket's configurations

Reply:
updated to dict